src/WebJobs.Script/Rpc/FunctionRegistration/FunctionDispatcher.cs

```diff
@@ -90,27 +95,44 @@ public LanguageWorkerState CreateWorkerStateWithExistingChannel(string language,
             return state;
         }
 
-        public LanguageWorkerState CreateWorkerState(string language)
+        public async Task InitializeAsync(string workerRuntime, IEnumerable<FunctionMetadata> functions)
         {
-            ILanguageWorkerChannel initializedChannel = _languageWorkerChannelManager.GetChannel(language);
+            workerRuntime = workerRuntime ?? Utility.GetWorkerRuntime(functions);
+            if (Utility.ShutdownStandbyChannels(_environment, functions, workerRuntime))
```

Comment:
nit: ShouldShutdownStandbyChannels? This method name makes it sound like it is the one doing the shutting down

Reply:
Also, I think it would simplify things if you just called _languageWorker.ShutdownStandbyChannelsAsync() - internally can't it do whatever checks it needs to do and do the right thing, w/o this double check here?